### PR TITLE
Enable CIS benchmark hardening in AIO and Multinode

### PR DESCRIPTION
**Pull request by jovial**
Currently we do not have coverage of this feature in CI.

### CODE REVIEW

File: doc/source/configuration/security-hardening.rst [hunk 1]
```diff
@@ -40,3 +40,19 @@ whether or not workloads or API requests are affected by any configuration chang
 
     kayobe playbook run $KAYOBE_CONFIG_PATH/ansible/cis.yml
 
+Enabling the host configure hook
```
Comment (markgoddard) [inline doc/source/configuration/security-hardening.rst:65]: Can you also document the new group?

File: etc/kayobe/hooks/overcloud-host-configure/post.d/99-cis.yml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+---
+
+- import_playbook: ../../../ansible/cis.yml
+  when: stackhpc_enable_cis_benchmark_hardening | bool
```
Comment (markgoddard) [inline etc/kayobe/hooks/overcloud-host-configure/post.d/99-cis.yml]: Needs a default setting somewhere.

Comment (markgoddard) [inline etc/kayobe/hooks/overcloud-host-configure/post.d/99-cis.yml]: Also it's not clear from the name that this only affects the hook.

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: etc/kayobe/hooks/overcloud-host-configure/post.d/99-cis.yml (hunk 2 above)
Comment (jovial) [reply etc/kayobe/hooks/overcloud-host-configure/post.d/99-cis.yml]: Fair point. I will move it into the playbook.

Comment (jovial) [reply etc/kayobe/hooks/overcloud-host-configure/post.d/99-cis.yml]: I did add a group_var: https://github.com/stackhpc/stackhpc-kayobe-config/pull/1104/files#diff-ec9ce8807b82c7b329b19049d68530c99cb7298a92298c7b534df085f24cf2faR6, unless I've misunderstood.

Comment (jovial) [reply etc/kayobe/hooks/overcloud-host-configure/post.d/99-cis.yml]: I changed my mind as that is technically a breaking change and updated the name of the variable.

File: doc/source/configuration/security-hardening.rst (hunk 1 above)
Comment (jovial) [reply doc/source/configuration/security-hardening.rst:65]: done

File: .github/workflows/stackhpc-all-in-one.yml [hunk 3]
```diff
@@ -355,6 +355,19 @@ jobs:
...
+        # we can lose the IP address that allows us to connect to the instances. This playbook
+        # also fixes that issue.
+        run: |
+          docker run -t --rm \
+            -v $(pwd):/stack/kayobe-automation-env/src/kayobe-config \
+            -e KAYOBE_ENVIRONMENT -e KAYOBE_VAULT_PASSWORD -e KAYOBE_AUTOMATION_SSH_PRIVATE_KEY \
+            ${{ steps.kayobe_image.outputs.kayobe_image }} \
+            /stack/kayobe-automation-env/src/kayobe-config/.automation/pipeline/playbook-run.sh etc/kayobe/ansible/configure-aio-resources.yml
+        env:
+          KAYOBE_AUTOMATION_SSH_PRIVATE_KEY: ${{ steps.ssh_key.outputs.ssh_key }}
```
Comment (markgoddard) [inline .github/workflows/stackhpc-all-in-one.yml:369]: I assume it's only required on upgrade?

Comment (jovial) [reply .github/workflows/stackhpc-all-in-one.yml:369]: ahh, good call

File: etc/kayobe/ansible/cis.yml [hunk 4]
```diff
@@ -1,8 +1,10 @@
 ---
 
 - name: Security hardening
-  hosts: overcloud
+  hosts: cis-hardening
```
Comment (markgoddard) [inline etc/kayobe/ansible/cis.yml:4]: Should the group_vars file be moved from overcloud to cis-hardening?

**Review by markgoddard — CHANGES_REQUESTED** [review]

File: etc/kayobe/ansible/cis.yml (hunk 4 above)
Comment (jovial) [reply etc/kayobe/ansible/cis.yml:4]: Good call - I've now moved them

**Review by markgoddard — APPROVED** [review]
Nice work

**Comment by markgoddard** [comment]
hmm, did I merge that by accident? The RL9 upgrade job failed due to being out of disk space in lv_docker:

`/dev/mapper/rootvg-lv_docker    32G   30G  188M 100% /var/lib/docker`

**Pull request merged**